src/Http/Http.Extensions/test/ProblemDetailsDefaultWriterTest.cs

```diff
@@ -54,6 +54,36 @@ public async Task WriteAsync_Works()
         Assert.Equal(expectedProblem.Instance, problemDetails.Instance);
     }
 
+    [Fact]
+    public async Task WriteAsync_Works_ProperCasing()
+    {
+        // Arrange
+        var writer = GetWriter();
+        var stream = new MemoryStream();
+        var context = CreateContext(stream);
+        var expectedProblem = new ProblemDetails()
+        {
+            Detail = "Custom Bad Request",
+            Instance = "Custom Bad Request",
+            Status = StatusCodes.Status400BadRequest,
+            Type = "https://tools.ietf.org/html/rfc9110#section-15.5.1-custom",
+            Title = "Custom Bad Request",
+        };
+        var problemDetailsContext = new ProblemDetailsContext()
+        {
+            HttpContext = context,
+            ProblemDetails = expectedProblem
+        };
+
+        //Act
+        await writer.WriteAsync(problemDetailsContext);
+
+        //Assert
+        stream.Position = 0;
+        var result = await JsonSerializer.DeserializeAsync<Dictionary<string, object>>(stream, SerializerOptions);
+        Assert.Equal(result.Keys, new(new() { { "type", 0 }, { "title", 1 }, { "status", 2}, { "detail", 3 }, { "instance", 4 } }));
```

Comment:
Should we add a test for "extensions" and "errors" (for HttpValidationProblemDetails and ValidationProblemDetails)?

Reply:
Can do!